tlv/record.go

```diff
@@ -12,9 +12,10 @@ import (
 // Type is an 64-bit identifier for a TLV Record.
 type Type uint64
 
-// TypeSet is an unordered set of Types. The map item boolean values indicate
-// whether the type that we parsed was known.
-type TypeSet map[Type]bool
+// TypeSet is an unordered set of Types. The map values are byte slices. If the
+// byte slice is nil, the type was successfully parsed. Otherwise the value is
+// byte slice containing the encoded data.
+type TypeSet map[Type][]byte
```

Comment:
This seems to be really overloading the type beyond its original intended usage. It's only meant to be used in order to determine violations w.r.t known/unknown types. We already return the full set of unparsed types after decoding, so not sure why this is needed.

Reply:
Also have you benched marked this? As is, it appears we'll end up copying all the values from the main stream _twice_.

Reply:
Yes it is overloading the type, but we can deviate from an original intention. 

I haven't benchmarked. If we want to process the raw bytes of unrecognized records, we need to copy anyway. Just in case our intention is to really ignore them, the copy is unnecessary. Afaik, for db tlv that isn't expected. And for the tlv payload it may happen with an unrecognized standard field (the custom ones we probably want to store somewhere). To me it seems premature to optimize for.

I am open to different ways of implementing this. What do you prefer?

Reply:
Where do you see the copy twice btw? Afaik we only read bytes from the stream and store them in a new byte slice that is a value of the `TypeSet` map.

Reply:
Renamed to `TypeMap` and preallocated the buffer.